pandas/_typing.py

```diff
@@ -1,4 +1,12 @@
 from pathlib import Path
-from typing import IO, AnyStr, Union
+from typing import IO, AnyStr, Type, Union
 
+import numpy as np
+
+from pandas.core.dtypes.dtypes import ExtensionDtype
+from pandas.core.dtypes.generic import ABCExtensionArray
+
+ArrayLike = Union[ABCExtensionArray, np.ndarray]
+SparseDtype = Union[str, np.dtype, ExtensionDtype,
```

Comment:
I think maybe we should call this something else, as SparseDtype is an actual type that we have (see my comment below)

cc @TomAugspurger 

alternatively, maybe we just move this particular type definition into pandas/core/arrays.py and don't put it here (as I am not sure we are going to be using this particular type very often).

Reply:
I think I misunderstood the comment above, then:
> I think we need to use SparseDtype here explicity here
Did you mean inside the `Union` as part of the definition itself?

Reply:
Might have missed the conversation but we just want to define Dtype here no?

Agreed SparseDtype in particular will have pretty limited usage, so not something I think we need to expose in this module

Reply:
yeah i think we can make this only Dtype (so remove the Type references ) 

these are generic types so we want to use in many places (eg in pandas.core.dtypes.dtypes)

Reply:
Just so I'm on the right page after all the discussion above, does rolling back to the [first commit](https://github.com/pandas-dev/pandas/pull/25943/commits/0761dabd30acebdfc9b9f0394980ffc90f9e6ae2) get us where we need to be?

That would return _typing.py to defining Dtype as is it currently defined in the existing type annotation, remove the specific Type references that were added after discussion above (`Type[float]`, `Type[int]`, `Type[object]`), and replace them with plain (and potentially over-broad) `Type`:
```py
Dtype = Union[str, np.dtype, ExtensionDtype, Type]
```

Alternatively, do we want to go back to Dtype and drop **all** the `Type`s? (This is how I interpret @jreback's "so remove the Type references" comment immediately above.) That would leave
```py
Dtype = Union[str, np.dtype, ExtensionDtype]
```
I don't think this change would comply with the documented functionality of SparseDtype's init method that we pulled the definition from, but it does satisfy mypy.

Reply:
The former should work here

Reply:
Does that work for you, @jreback?

Reply:
@WillAyd for `Dtype = Union[str, np.dtype, ExtensionDtype, Type]`
@jreback for `Dtype = Union[str, np.dtype, ExtensionDtype]`

Both satisfy mypy. I propose we pick the more restrictive one: `Dtype = Union[str, np.dtype, ExtensionDtype]`.

Reply:
The only reason I was thinking we might need Type is because something like `dtype=float` is valid. OK to start smaller for now though - can add in later when it comes up